alacritty/src/display/mod.rs

```diff
@@ -964,8 +996,12 @@ impl Display {
 
         // Update cursor shape.
         if highlighted_hint.is_some() {
+            // If mouse changed the line, we'd need to update the preview.
+            dirty = self.hint_mouse_point.map(|p| p.line != point.line).unwrap_or(false);
+            self.hint_mouse_point = Some(point);
             self.window.set_mouse_cursor(CursorIcon::Hand);
```

Comment:
This is accurate, since the only thing we care about is line change, the rest and hint change is being handled earlier in this function.

The particular change was added to catch a situation when the hint stays the same,  but cursor changes the line, so we can change URL preview location if it obscures anything.

Reply:
You're specifically talking about the mouse though, right? And the obfuscation only checks for obfuscation with the cursors? So a mouse change could never introduce a change unless the "mouse" is the vi mode cursor which clearly isn't a mouse?

Reply:
No. the mouse cursor could also be obfuscated, that's the point. If you have terminal of height 2 and have the same hyperlink on both lines starting highlight on first line will result in showing preview in the second line, however if you move to the second line preview will still appear on the second line, since the hint wasn't change, because hint occupies multiple lines and haven't changed, only cursor position was changed in that case.

Reply:
The mouse cursor is *above* the terminal, it's impossible for hints to render above the cursor. I don't see why the mouse cursor's location would be relevant at all, what you're saying sounds like you're talking about the highlighted URL, not the mouse cursor.

Reply:
I mean that preview will obscure the link, not the cursor as a visual object in particular. Right, I meant to say that whatever we highlight with mouse cursor won't be visible. So we should move preview to show that the link is clickable, since preview isn't meant to be clickable.

Reply:
That has nothing to do with the mouse then and should be more clearly documented.